packages/admin/admin/src/form/FinalFormFileUpload.tsx

```diff
@@ -0,0 +1,167 @@
+            height: "80px",
+            padding: "10px",
+            justifyContent: "center",
+            alignItems: "center",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            border: `1px dashed ${palette.text.secondary}`,
+            cursor: "pointer",
+            minWidth: "250px",
+            "&$disabled": {
+                cursor: "default",
+            },
+            "&$error": {
+                border: `1px dashed ${palette.error.main}`,
+            },
+        },
+        dropableAreaCaption: {
+            color: palette.text.secondary,
+            fontSize: "14px",
+            fontStyle: "normal",
+            fontWeight: "300",
+            lineHeight: "20px",
+        },
+        fileList: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flex-start",
+            gap: "2px",
+            alignSelf: "stretch",
+        },
+        fileListItem: {
+            display: "flex",
+            height: "35px",
+            padding: "8px 7px 8px 15px",
+            alignItems: "center",
+            gap: "10px",
+            alignSelf: "stretch",
+            borderRadius: "4px",
+            background: palette.background.default,
+            justifyContent: "space-between",
+            minWidth: "250px",
+        },
+        disabled: {},
+        error: {},
+    });
+};
+
+export interface FinalFormFileUploadProps extends FieldRenderProps<HTMLInputElement> {
+    dropzoneVariant: "dropzoneOnly" | "buttonOnly" | "default";
+    caption: React.ReactNode;
+    multipleFiles: boolean;
+    accept: Accept;
+}
+
+const FinalFormFileUploadComponent: React.FunctionComponent<WithStyles<typeof styles> & FinalFormFileUploadProps> = ({
+    classes,
+    disabled,
+    dropzoneVariant,
+    caption,
+    multipleFiles = true,
+    accept,
+    input: { name, onChange, value: fieldValue },
+}) => {
+    // convert bytes for the file size chip
+    function formatBytes(bytes: number, decimals = 2) {
+        if (!+bytes) return "0 Bytes";
+
+        const k = 1024;
+        const dm = decimals < 0 ? 0 : decimals;
+        const sizes = ["Bytes", "KB", "MB", "GB"];
+
+        const i = Math.floor(Math.log(bytes) / Math.log(k));
+
+        return `${parseFloat((bytes / Math.pow(k, i)).toFixed(dm))} ${sizes[i]}`;
+    }
+
+    // add myFiles state to be able to remove accepted files
+    const [myFiles, setMyFiles] = React.useState<File[]>([]);
+
+    const onDrop = React.useCallback(
+        (acceptedFiles: File[]) => {
+            setMyFiles([...myFiles, ...acceptedFiles]);
+        },
+        [myFiles],
+    );
+
+    const removeFile = (file: File) => () => {
+        const newFiles = [...myFiles];
+        newFiles.splice(newFiles.indexOf(file), 1);
+        setMyFiles(newFiles);
+    };
+
+    const { acceptedFiles, getRootProps, getInputProps, isDragReject } = useDropzone({ onDrop, accept, disabled, multiple: multipleFiles });
```

Comment:
Currently, you don't handle the case of a rejected file. You should at least show a generic error state.

Reply:
Good point, I added it today.